keydata.go

```diff
@@ -302,6 +313,99 @@ func (d *KeyData) snapModelAuthKey(auxKey AuxiliaryKey) ([]byte, error) {
 	return hmacKey, nil
 }
 
+func (d *KeyData) updatePassphrase(payload, oldKey []byte, passphrase string, kdfOptions *KDFOptions, kdf KDF) error {
+	handler := handlers[d.data.PlatformName]
+	if handler == nil {
+		return ErrNoPlatformHandlerRegistered
+	}
+
+	if kdfOptions == nil {
+		var defaultOptions KDFOptions
+		kdfOptions = &defaultOptions
+	}
+
+	keyLen := passphraseDerivedKeyLen + aes.BlockSize
```

Comment:
this probably merits a comment about why this is chosen this way

Reply:
I've added a comment to this now.